app/src/main/java/org/schabi/newpipe/player/MainVideoPlayer.java

```diff
@@ -1031,6 +1087,21 @@ public void showControls(final long duration) {
             super.showControls(duration);
         }
 
+        @Override
+        public void safeHideControls(final long duration, final long delay) {
+            if (DEBUG) {
+                Log.d(TAG, "safeHideControls() called with: delay = [" + delay + "]");
+            }
+
+            View controlsRoot = getControlsRoot();
+            if (controlsRoot.isInTouchMode()) {
+                getControlsVisibilityHandler().removeCallbacksAndMessages(null);
+                getControlsVisibilityHandler().postDelayed(() ->
+                        animateView(controlsRoot, false, duration, 0,
+                                MainVideoPlayer.this::hideSystemUi), delay);
+            }
+        }
+
```

Comment:
I have a question: why was this added? It is causing trouble with controls not being hidden after pausing and resuming (See #3403)
@TobiGr @Alexander--